### PR TITLE
TSan Changes for QEMU

### PR DESCRIPTION
**Pull request by rf972**

### CODE REVIEW

File: accel/tcg/tcg-all.c [hunk 1]
```diff
@@ -54,8 +54,8 @@ static void tcg_handle_interrupt(CPUState *cpu, int mask)
     int old_mask;
     g_assert(qemu_mutex_iothread_locked());
 
-    old_mask = cpu->interrupt_request;
-    cpu->interrupt_request |= mask;
+    old_mask = atomic_read(&cpu->interrupt_request);
+    atomic_set(&cpu->interrupt_request, cpu->interrupt_request | mask);
```
Comment (PeterPuhov-Linaro) [inline accel/tcg/tcg-all.c:58]: You reading old_mask, bur ORing with mask.
Would it be better to use atomic OR ?

File: hw/core/cpu.c [hunk 2]
```diff
@@ -104,7 +104,7 @@ void cpu_reset_interrupt(CPUState *cpu, int mask)
     if (need_lock) {
         qemu_mutex_lock_iothread();
     }
-    cpu->interrupt_request &= ~mask;
+    atomic_set(&cpu->interrupt_request, cpu->interrupt_request & ~mask);
```
Comment (PeterPuhov-Linaro) [inline hw/core/cpu.c:107]: Would it be better to use atomic_and here?

**Comment by lain3d** [comment]
what's the state of this? any problems?

**Comment by rf972** [comment]
The issues in this PR have all been resolved.  The patch is in good shape and the first revision will be submitted up to the qemu-devel mailing list soon.

**Pull request closed without merging**